core/src/main/java/org/apache/iceberg/rest/responses/FetchPlanningResultResponseParser.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.responses;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.List;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.PlanStatus;
+import org.apache.iceberg.util.JsonUtil;
+
+public class FetchPlanningResultResponseParser {
+  private static final String PLAN_STATUS = "plan-status";
+  private static final String PLAN_TASKS = "plan-tasks";
+
+  private FetchPlanningResultResponseParser() {}
+
+  public static String toJson(FetchPlanningResultResponse response) {
+    return toJson(response, false);
+  }
+
+  public static String toJson(FetchPlanningResultResponse response, boolean pretty) {
+    return JsonUtil.generate(gen -> toJson(response, gen), pretty);
+  }
+
+  public static void toJson(FetchPlanningResultResponse response, JsonGenerator gen)
+      throws IOException {
+    Preconditions.checkArgument(
+        null != response, "Invalid response: fetchPanningResultResponse null");
+    gen.writeStartObject();
+    gen.writeStringField(PLAN_STATUS, response.planStatus().status());
+    if (response.planTasks() != null) {
+      JsonUtil.writeStringArray(PLAN_TASKS, response.planTasks(), gen);
+    }
+
+    TableScanResponseParser.serializeScanTasks(
+        response.fileScanTasks(), response.deleteFiles(), response.specsById(), gen);
+    gen.writeEndObject();
+  }
+
+  public static FetchPlanningResultResponse fromJson(String json) {
+    Preconditions.checkArgument(json != null, "Invalid response: fetchPanningResultResponse null");
+    return JsonUtil.parse(json, FetchPlanningResultResponseParser::fromJson);
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
I think we can remove this now since we dont have so many conditionals in the method.